### ISSUE TITLE
Add support for parsing composite merge commit messages

### ISSUE THREAD

**Issue by flelli** [mooltiverse/nyx#125]
So far the tools uses commit message conventions to infer the commit type and the identifiers to bump based on the header of a commit message.

Merge commits, though, usually have commit messages that list all commits belonging to the merge/pull request in the message body (in addition to the merge commit header).

It's useful to infer the commit type and bump not just on the message header but also on the lines within the message body (at least those matching the commit message convention) so that identifiers to bump can be directly inferred by all commits belonging to the merge, selecting the most significant one.

This can be enabled as a flag on the release type so that it can be turned on/off based on the target branch.

Chances are that this feature only works for merge commits (those with more than one parent commit).

**Comment by riccardopinosio** [comment]
Hello! Indeed this feature would be super useful. Currently we are in the situation you're describing, we have a master branch and a development branch off of which we create feature branches. Then we regularly merge development into master after testing. In this case, the merge creates a merge commit on master, and the changelog only reflects what is in the title of the merge commit. It would be great to have it reflect also based on the lines of the message body.

**Comment by flelli** [comment]
So relaxing the commit message [expression](https://mooltiverse.github.io/nyx/guide/user/configuration-reference/commit-message-conventions/#expression) to have multiple matches across multiple lines should do the job. The expression:

```
(?<type>[a-zA-Z0-9_]+)(!)?(\((?<scope>[a-z ]+)\))?:( (?<title>.+)).*
```

has been tested [here](https://regex101.com/r/mTuzqu/1) against an example set of commits with messages:

1. `fix: patch 1`
2. `feat: minor 1`
3. `feat!: major 1`
4. `fix: patch 2`
5. `feat: minor 2`
6. `feat!: major 2`

These commits, when merge (with squash) on GitHub from an `alpha` branch, result in a message like:

```
Alpha (#1)

* fix: patch #1

* feat: minor #1

* feat!: major #1

* fix: patch #2

* feat: minor #2

* feat!: major #2
```

and the above expression indeeds matches 6 times. The expression still matches the header (and all the lines in the body) when the header is compliant with conventional commits.

Of course this is not to be considered a **standard** conventional commits, but can be delivered as a custom convention for those using conventional commits and willing to parse composite messages generated for pull/merge requests.

The changelog generation will probably need some code changes to also include commit messages from the body.

**Comment by flelli** [comment]
Hi @riccardopinosio and @lxke19 ,

this feature is now available in [2.5.1](https://github.com/mooltiverse/nyx/releases/tag/2.5.1).

There is a new commit message convention named `conventionalCommitsForMerge` in the 
 [extended preset](https://mooltiverse.github.io/nyx/guide/user/configuration-presets/extended/) which does what you need. Since it's not official and could break backward compatibility it's not enabled by default so you need to [enable](https://mooltiverse.github.io/nyx/guide/user/configuration-reference/commit-message-conventions/#enabled) it yourself.

If you're not using the _extended_ preset you can configure the convention in your custom configuration file using:

- [`expression`](https://mooltiverse.github.io/nyx/guide/user/configuration-reference/commit-message-conventions/#expression) = `(?<type>[a-zA-Z0-9_]+)(!)?(\\((?<scope>[a-z ]+)\\))?:( (?<title>.+))`
- [`bump-expressions`](https://mooltiverse.github.io/nyx/guide/user/configuration-reference/commit-message-conventions/#bump-expressions):
  - `major` = `(?s)(?m)[a-zA-Z0-9_]+(!: .*|.*^(BREAKING( |-)CHANGE: )).*`
  - `minor` = `(?s)(?m)feat(!{0})(\\([a-z ]+\\))?: (?!.*^(BREAKING( |-)CHANGE: )).*`
  - `major` = `(?s)(?m)fix(!{0})(\\([a-z ]+\\))?: (?!.*^(BREAKING( |-)CHANGE: )).*`

Note backslashes are already escaped.

See the [configuration reference](https://mooltiverse.github.io/nyx/guide/user/configuration-reference/commit-message-conventions/) for more.

**Comment by lxke19** [comment]
We really appreciate this, thank you!
